### ISSUE TITLE
conda version 4.1.4 breaks git-info for zsh

### ISSUE THREAD

**Issue by michaelaye** [conda/conda#3128]
After updating conda from 4.1.3 to 4.1.4 (I retraced this thanks to conda's easy version management! ;)), the `git-info` functionality for `zsh` is broken.
Here's how things look using conda 4.1.3 for a `cd` into a git folder:

``` zsh
┌─( ~ )───────────────────────────────────────────────────────────────────( @MacL2938 )─┐
└─❱❱❱ cdp4                                                                +4600 17:01 ❰─┘
LICENSE                   data                      planet4.sublime-workspace
MANIFEST.in               dist                      requirements.txt
Planet4.egg-info          meg                       setup.cfg
README.md                 notebooks                 setup.py
__pycache__               planet4                   tests
build                     planet4.sublime-project

┌─( ~/D/s/planet4 λ:master | ✱ ◼ f91418b )────────────────────────────────( @MacL2938 )─┐
└─❱❱❱ source activate stable                                              +4601 17:01 ❰─┘
prepending /Users/klay6683/miniconda3/envs/stable/bin to PATH

┌─( ~/D/s/planet4 λ:master | ✱ ◼ f91418b )────────────────────────────────( @MacL2938 )─┐
(stable) └─❱❱❱ conda --version                                            +4602 17:02 ❰─┘
conda 4.1.3

┌─( ~/D/s/planet4 λ:master | ✱ ◼ f91418b )────────────────────────────────( @MacL2938 )─┐
(stable) └─❱❱❱                                                            +4603 17:02 ❰─┘
```

Here's how it looks with conda 4.1.4:

``` zsh
┌─( ~ )───────────────────────────────────────────────────────────────────( @MacL2938 )─┐
└─❱❱❱ cdp4                                                                +4606 17:05 ❰─┘
LICENSE                   data                      planet4.sublime-workspace
MANIFEST.in               dist                      requirements.txt
Planet4.egg-info          meg                       setup.cfg
README.md                 notebooks                 setup.py
__pycache__               planet4                   tests
build                     planet4.sublime-project

┌─( ~/D/s/planet4 λ:master | ✱ ◼ f91418b )────────────────────────────────( @MacL2938 )─┐
└─❱❱❱ source activate stable                                              +4607 17:05 ❰─┘
prepending /Users/klay6683/miniconda3/envs/stable/bin to PATH
git-info:165: bad math expression: operator expected at `0'
git-info:173: bad math expression: operator expected at `0'

┌─( ~/D/s/planet4 λ:master | ✱ ◼ f91418b )────────────────────────────────( @MacL2938 )─┐
(stable) └─❱❱❱ conda --version                                            +4608 17:05 ❰─┘
conda 4.1.4
git-info:165: bad math expression: operator expected at `0'
git-info:173: bad math expression: operator expected at `0'

┌─( ~/D/s/planet4 λ:master | ✱ ◼ f91418b )────────────────────────────────( @MacL2938 )─┐
(stable) └─❱❱❱                                                            +4609 17:05 ❰─┘
```

During testing, I made sure that I complete restart my terminal. I did that because the above git-info error symptom remains even after a `source deactivate`!

The `zsh` community had a similar problem 2 years ago, see sorin-ionescu/prezto#566.
There the last comment indicates the culprit being a "virtualenv postactivate script which modified IFS and didn't unset it when finished.", which I admit sounds like Mandarin to me. ;) Could this be a similar issue?

Any help with this would be appreciated, as I need my zsh power-prompt `garrett` and conda to be friends! ;)

**Comment by michaelaye** [comment]
Even a complete reinstall of my conda setup did not remove this error, so this is real and not a setup remnant.

**Comment by michaelaye** [comment]
Just a hypothesis: Looking at the comparison of conda 4.1.3..4.1.4:
https://github.com/conda/conda/compare/4.1.3...4.1.4
I see that `CONDA_PREFIX` is replacing more that one thing, at the beginning, it replaces `CONDA_ENV_PATH` only, but then also `CONDA_DEFAULT_ENV`?
It's also confusing to find statements like "CONDA_PREFIX is always the full path" as that's not what usually a PREFIX is indicating?
In shell/conda.fish specifically (I don't know if that could block git-info for zsh, but as the old issue mentions `didn't unset`, I thought a problem with deactivation scripts rings the bell:
In lines 146 to 155 in that script, i find it weird that one CONDA_DEFAULT_ENV is replaced by CONDA_PREFIX and the other one not (lines 147 vs 149)?

I'm really stabbing in the dark here, hoping that it could point someone in the right direction, sometimes stabbing in the dark can catch you a fish (or kill an innocent bystander...).

**Comment by michaelaye** [comment]
hold off with getting active here @kalefranz , I now have a setup where this error does not appear, despite `conda info` giving the exact same things. Must find out the difference...

**Comment by michaelaye** [comment]
I reinstalled the full conda setup, including trashing the .conda folder (and closing the terminal in case there was something set in env variables I didn't see), and finally this bug goes away. Annoying, but thanks to conda being awesome, not such a biggie. Plus, it helps to have a file with the standard packages I install.

**Comment by michaelaye** [comment]
Error came back, after a bit of working in terminal. In both Terminal.app and iTerm2(v3.x), so I do think this is conda related.

**Comment by mingwandroid** [comment]
May be fixed by https://github.com/conda/conda/commit/c4bd2279e68af495d7a3bd60c3a03b15cdbc2318

**Comment by michaelaye** [comment]
I am currently fixing it with simply do ```unset IFS``` after the IFS definition lines, both in the `activate` and the `deactivate` script. I thought this was one of the PR merged into https://github.com/conda/conda/pull/3175 but maybe I'm wrong.

**Comment by mingwandroid** [comment]
`activate` and `deactivate` do not use IFS anymore: https://github.com/conda/conda/blob/4.3.x/shell/activate https://github.com/conda/conda/blob/4.3.x/shell/deactivate

https://github.com/conda/conda/pull/3175 hasn't been merged into the 4.3 branch.

**Comment by michaelaye** [comment]
i see. i'm on conda-forge, that's why i didn't see this yet, they use 4.2 still.
